src/pages/gen2/build-a-backend/auth/auth-triggers/index.mdx

```diff
@@ -0,0 +1,90 @@
+};
+
+export function getStaticProps(context) {
+  return {
+    props: {
+      meta
+    }
+  };
+}
+
+<Callout info>
+
+**Coming soon:** The Functions experience in Amplify (Gen 2) is still under development. Expect the API surface to change dramatically during the Gen 2 preview period. If you have any feedback on the desired developer experience, please [file a GitHub issue](https://github.com/aws-amplify/amplify-backend/issues).
+
+</Callout>
+
+[Amazon Cognito Triggers](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-identity-pools-working-with-aws-lambda-triggers.html) enable you to customize authentication flows using [AWS Lambda](https://aws.amazon.com/lambda/) functions. Customizations such as allowlisting email domains or creating a `UserProfile` model after sign-up can be written to fit your needs.
+
+If you have not yet created an Amplify (Gen 2) app, visit the [quickstart](/gen2/start/quickstart).
+
+## Creating your first trigger
+
+To get started with your first trigger, create a file at `amplify/auth/pre-signup/index.js` with the following contents:
+
+```js title="amplify/auth/pre-signup/index.js"
+// amplify/auth/pre-signup/index.js
+/**
+ * My pre-signup trigger
+ * @type {import('aws-lambda').PreSignUpTriggerHandler}
+ */
+exports.handler = async (event) => {
+  console.log('event: ', JSON.stringify(event))
+  return event
+}
+```
+
+<Callout info>
+
+This snippet includes JSDoc to type the function, including its event and return type. Install the [`@types/aws-lambda`](https://www.npmjs.com/package/@types/aws-lambda) package by using the following command:
+
+```bash
+npm add --save-dev @types/aws-lambda
+```
+
+</Callout>
+
+Then, using the experimental `Func` module, configure your authentication resource to use the newly-created trigger file:
+
+```ts title="amplify/auth/resource.ts"
+// amplify/auth/resource.ts
+import { defineAuth, Func } from '@aws-amplify/backend'
+
+export const auth = defineAuth({
+  loginWith: {
+    email: true,
+  },
+  triggers: {
+    // use `Func.fromDir` to use the relative directory of the created trigger
+    preSignUp: Func.fromDir({
+      name: 'pre-signup',
+      codePath: './pre-signup',
+    }),
+  },
+})
+```
+
+`Func.fromDir` enables you to author Functions in any runtime by specifying an optional `runtime` property. For example, if you have a Python function specify the Python runtime from the [AWS Cloud Development Kit (CDK)](https://docs.aws.amazon.com/cdk/):
+
+```diff title="amplify/auth/resource.ts"
+// amplify/auth/resource.ts
++import { Runtime } from 'aws-cdk-lib/aws-lambda'
+import { defineAuth, Func } from '@aws-amplify/backend'
+
+export const auth = defineAuth({
+  loginWith: {
+    email: true,
+  },
+  triggers: {
+    // use `Func.fromDir` to use the relative directory of the created trigger
+    preSignUp: Func.fromDir({
+      name: 'pre-signup',
+      codePath: './pre-signup',
++     runtime: Runtime.PYTHON_3_9,
```

Comment:
Let's address this by moving it into the functions docs. Instead of with triggers.